### PR TITLE
Add test case for internal/errors/discoverer.go

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added the test cases for `internal/errors/discoverer.go` and comments.

***grammar check passed***

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errors/discoverer_test.go [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns discover pod not found error when ns is `vald-discoverer`",
+			args: args{
+				name: "vald-discovere",
```
Comment (kevindiu) [inline internal/errors/discoverer_test.go:239]:
```suggestion
				name: "vald-discoverer",
```

**Review by kevindiu — DISMISSED** [review]
LGTM

**Review by kevindiu — APPROVED** [review]
LGTM

File: internal/errors/discoverer_test.go [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns wrapped discover error when err is server error and record is `127.0.0.1`",
```
Comment (vankichi) [inline internal/errors/discoverer_test.go:29]:
```suggestion
			name: "returns wrapped discover error when err is server error and record is '127.0.0.1'",
```

File: internal/errors/discoverer_test.go [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+			args: args{
+				err:    New("server error"),
+				record: "",
+			},
+			want: want{
+				want: New("addr  ip couldn't discover: server error"),
+			},
+		},
+		{
+			name: "returns discover error when err is nil error and record is `127.0.0.1`",
```
Comment (vankichi) [inline internal/errors/discoverer_test.go:49]:
```suggestion
			name: "returns discover error when err is nil error and record is '127.0.0.1'",
```

File: internal/errors/discoverer_test.go [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns discover node not found error when node is `127.0.0.1`",
```
Comment (vankichi) [inline internal/errors/discoverer_test.go:113]:
```suggestion
			name: "returns discover node not found error when node is '127.0.0.1'",
```

File: internal/errors/discoverer_test.go [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns discover namespace not found error when ns is `vald`",
```
Comment (vankichi) [inline internal/errors/discoverer_test.go:175]:
```suggestion
			name: "returns discover namespace not found error when ns is 'vald'",
```

File: internal/errors/discoverer_test.go [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "returns discover pod not found error when ns is `vald-discoverer`",
```
Comment (vankichi) [inline internal/errors/discoverer_test.go:237]:
```suggestion
			name: "returns discover pod not found error when ns is 'vald-discoverer'",
```

**Review by vankichi — COMMENTED** [review]
I recommend you to use '' instead of `` when you'd like to use the symbol for wrapping the actual value in each test case name.
If you fix, LGTM

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request merged**